### DISCUSSION TITLE
HOGWARTS: Duplicates #61 My sort button isn't working with the hide form function

### DISCUSSION THREAD

**Discussion by elliewalkersm** [nss-evening-cohort-14/evening-client-side#70] [Help Me!] [answered]
Have you searched the other help tickets? If not, do that first (Duplicates asks for help will be shamed ceaselessly)
yes

Describe your issue
My sort button was working fine when I had the form on my HTML page, but once I moved the form onto my js page in a showForm function the sort button no longer works.

List any error messages that appear
Uncaught TypeError: Cannot read property 'addEventListener' of null
    at buttonClick (main.js:76)
    at init (main.js:80)
    at main.js:84

What have you tried? Tell us everything you have tried.
I took a step back and commented out the new showForm function and added the form back to my HTML and the sort button works. I tried putting the showForm function back in and the sort button stopped working again. I thought it might be because there is no longer an id="sortBtn" on my HTML page which is being targeted in my buttonClick function

Copy/paste link TO THE FILE on your branch or in your repo that you are having issues with https://github.com/elliewalkersm/sorting-hat/blob/hide-form/main.js

**Comment by TrinityChristiana** [comment]
This was answered in this ticket: https://github.com/nss-evening-cohort-14/evening-client-side/discussions/61. If this does not help you with your issue feel free to comment there.

**Reply by elliewalkersm** [reply]
My function is set up a little different so I'm still having trouble.  Instead of setting a variable for the form, I have an if statement that targets the form id and used .innerHTML.

**Comment by Gabrielle-Tobermann** [comment]
I am having the same issue (I think). Ever since I added my createForm function to show the form only when the button is clicked, my cards won't print to the DOM. I commented out all of my functions except createStudentCards, and it works fine. It stops working when the other functions are there. I wonder if it's a control flow issue but I can't tell. I am not getting any errors.

What I've tried: 
-commenting out everything except createStudentCards (it works on its own).
-in html, moving my div with id of 'form' under my div with if of 'students' , no changes. 
- I looked at pies again, but I couldn't see anything different. 

Here is my file: 
https://github.com/Gabrielle-Tobermann/sorting-hat/blob/delete-functionality/main.js

**Comment by drteresavasquez** [comment]
@TrinityChristiana is right. #61 will fix your issue. Make sure to take a look at what the solution was for it. You cannot target an element on the DOM until it exists. You will need to move your query selector and event listener to after the form loads in the DOM in the function you created.

**Comment by elliewalkersm** [comment, marked as answer]
I was targeting my form id before the form was built, so I moved the document.querySelector from the buttonClick function into my showForm function.

```
const showForm = (e) => {
  const buttonId = e.target.id;

  if (buttonId === "startBtn") {
    document.querySelector("#studentForm").innerHTML = `<div id="studentForm">
    <form>
      <h1>Enter First Year's Name</h1>
      <div class="row mb-3">
      <label for="student-name" class="col-sm-2 col-form-label">Student:</label>
      <div class="col-sm-10">
        <input type="text" class="form-control d-inline" id="studentName" required>
      </div>
      </div>
      <button type="button" class="btn btn-secondary" id="sortBtn">Sort!</button>
    </form>
  </div>`;
  }
  document.querySelector("#sortBtn").addEventListener("click", formContent);
};```

**Comment by drteresavasquez** [comment]
@Gabrielle-Tobermann what was your solution?
